From: Steve Best <sbest@redhat.com>
Date: Mon, 30 Aug 2010 15:17:41 -0400
Subject: [scsi] ipr: add writeq definition if needed
Message-id: <20100830150554.5212.23813.sendpatchset@squad5-lp1.lab.bos.redhat.com>
Patchwork-id: 27923
O-Subject: [PATCH RHEL5.6 BZ626566 9/16] ipr: add writeq definition if needed
Bugzilla: 626566

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=626566

Description:
------------
Compiling the driver will fail on 32 bit powerpc and other
architectures where writeq is not defined.  This patch adds a
definition for writeq.

Signed-off-by: Wayne Boyer <wayneb@linux.vnet.ibm.com>
Acked-by: Brian King <brking@linux.vnet.ibm.com>
Signed-off-by: James Bottomley <James.Bottomley@suse.de>

RHEL Version Found:
-------------------
5.6

Brew:
-----
Built on all platforms. All patches brewed together.
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=2713332

Upstream Status:
----------------
http://git.kernel.org/?p=linux/kernel/git/jejb/scsi-misc-2.6.git;
a=commit;h=c5f10187965f93ef7ef67da9c7c449b13b6dee1b

Steve Best
IBM on-site partner

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/ipr.h b/drivers/scsi/ipr.h
index fbef7de..74a4b77 100644
--- a/drivers/scsi/ipr.h
+++ b/drivers/scsi/ipr.h
@@ -1874,4 +1874,12 @@ static inline int ipr_sdt_is_fmt2(u32 sdt_word)
 #endif
 
 
+#ifndef writeq
+static inline void writeq(u64 val, void __iomem *addr)
+{
+	writel(((u32) (val >> 32)), addr);
+	writel(((u32) (val)), (addr + 4));
+}
 #endif
+
+#endif /* _IPR_H */
